src/DynamoCore/Models/AnnotationModel.cs

```diff
@@ -0,0 +1,379 @@
+
+                var maxWidth = widthandheight.Item1;
+                var maxHeight = widthandheight.Item2;
+
+                // InitialTop is to store the Y value without the Textblock height
+                this.InitialTop = groupModels.Min(y => y.Y);
+
+                var region = new Rect2D
+                {
+                    X = regionX,
+                    Y = regionY,
+                    Width = xDistance + maxWidth + 10,
+                    Height = yDistance + maxHeight + 10
+                };
+             
+                this.Left = region.X;              
+                this.Top = region.Y;
+                this.Width = region.Width;
+                this.Height = region.Height;
+
+                //Calculate the boundary if there is any overlap
+                ModelBase overlap = null;
+                foreach (var nodes in SelectedModels)
+                {
+                    if (!region.Contains(nodes.Rect))
+                    {
+                        overlap = nodes;
+                        if (overlap.Height < this.Height)
+                        {
+                            if (overlap.Rect.Top < this.Top ||
+                                    overlap.Rect.Bottom > region.Bottom) //Overlap in height - increase the region height
+                            {
+                                this.Height += overlap.Rect.Bottom - region.Bottom + 10;
+                                region.Height = this.Height;
+                            }
+                            if (overlap.Rect.Left < this.Left ||
+                                    overlap.Rect.Right > region.Right) //Overlap in width - increase the region width
+                            {
+                                this.Width += overlap.Rect.Right - region.Right + 10;
+                                region.Width = this.Width;
+                            }
+                        }
+                    }
+                }
+                               
+                //Initial Height is to store the Actual height of the group.
+                this.InitialHeight = region.Height;
+            }
+        }
+
+        /// <summary>
+        /// Group the Models based on Height and Width
+        /// </summary>
+        /// <returns></returns>
+        private Tuple<Double,Double> CalculateWidthAndHeight()
+        {           
+            var xgroup = SelectedModels.OrderBy(x => x.X).ToList();
+            var ygroup = SelectedModels.OrderBy(x => x.Y).ToList();
+          
+            return Tuple.Create(xgroup.Last().Width, ygroup.Last().Height);
+        }
+        
+        /// <summary>
+        /// Deserializes the model guids from XML
+        /// and creates group on those model guids.
+        /// </summary>
+        private void DeserializeGroup()
+        {
+            var listOfModels = new List<ModelBase>();
+            foreach (var objGuid in modelGuids.Split(','))
```

Comment:
This is a bit dubious.  You can store your modelguids in such a way that they can be deserialized as a collection, rather than having to do this kind of splitting.  That way the structure of your data is encoded in the XML directly.